File: binary.py
import math

def binary_search(list, item):
       left = 0
       right = len(list) - 1

       while (left <= right):
           mid = math.floor((left + right) / 2)
           guess = list[mid]

           if (guess == item):
               return mid 
           if (guess > item):
               right = mid - 1
           else:
               left = mid + 1 
        
       return None 

my_list = [1, 3, 7, 8, 44, 60, 334, 343434]
print(binary_search(my_list, 334))

File: binary.js
function binarySearch(arr, key) {
  let low = 0,
      high = arr.length - 1;

  while (low <= high) {
    let mid = Math.floor(low + (high - low) / 2),
        midVal = arr[mid];

    if (key === midVal) return mid;
    if (key > midVal) {
      low = mid + 1;
    } else {
      high = mid - 1;
    }
  }

  return -1;
}

function binarySearchDesc(arr, key) {
  let low = 0,
      high = arr.length - 1;

  while (low <= high) {
    let mid = Math.floor(low + (high - low) / 2),
        midVal = arr[mid];

    if (key === midVal) return mid;
    if (key > midVal) {
      high = mid - 1;
    } else {
      low = mid + 1;
    }
  }

  return -1;
}

module.exports = {
  binary: binarySearch,
  descBinary: binarySearchDesc
}
